middle_end/flambda2/types/grammar/type_grammar.ml

```diff
@@ -104,9 +118,9 @@ and head_of_kind_value =
    if we don't actually use the extensions, while the second version would be
    particularly useful if we switch several times on the same scrutinee. *)
 and head_of_kind_naked_immediate =
-  | Naked_immediates of Targetint_31_63.Set.t
-  | Is_int of t
-  | Get_tag of t
+  { immediates : Targetint_31_63.Set.t Or_unknown.t;
+    relations : RelationSet.t
+  }
```

Comment:
```suggestion
  | Naked_immediates of Targetint_31_63.Set.t
  | Relations of RelationSet.t
  | Both of
  { immediates : Targetint_31_63.Set.t Or_unknown.t;
    relations : RelationSet.t
  }
```
For compaction, as most of the time there would be only one of the two parts and to avoid an indirection for through Ok in the common case, this change might be benefitial (bu tedious to apply).